server/src/test/java/org/jetbrains/bsp/bazel/server/sync/languages/rust/RustWorkspaceResolverTest.kt

```diff
@@ -0,0 +1,307 @@
+import io.kotest.matchers.maps.shouldContainExactly
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.shouldNotBe
+import org.jetbrains.bsp.bazel.bazelrunner.BasicBazelInfo
+import org.jetbrains.bsp.bazel.bazelrunner.BazelRelease
+import org.jetbrains.bsp.bazel.server.sync.BazelPathsResolver
+import org.jetbrains.bsp.bazel.server.sync.model.*
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import java.net.URI
+import java.nio.file.Paths
+
+
+class RustWorkspaceResolverTest {
+
+  private val execRoot = "/private/var/tmp/_bazel/125c7a6ca879ed16a4b4b1a74bc5f27b/execroot/bazel_bsp"
+  private lateinit var bazelPathsResolver: BazelPathsResolver
+  private lateinit var resolver: RustWorkspaceResolver
+
+  @BeforeEach
+  fun beforeEach() {
+    // given
+    val bazelInfo = BasicBazelInfo(
+      execRoot = execRoot,
+      workspaceRoot = Paths.get("/Users/user/workspace/bazel-bsp"),
+      release = BazelRelease.fromReleaseString("release 6.0.0")
+    )
+
+    bazelPathsResolver = BazelPathsResolver(bazelInfo)
+    resolver = RustWorkspaceResolver(bazelPathsResolver)
+  }
+
+  private fun createModule(
+      label: String,
+      directDependencies: List<Label>,
+      rustModule: RustModule
+  ): Module =
+      Module(
+          label = Label(label),
+          isSynthetic = false,
+          directDependencies = directDependencies,
+          languages = setOf(Language.RUST),
+          tags = setOf(Tag.APPLICATION),
+          baseDirectory = URI.create("/path/to/base/directory"),
+          sourceSet = SourceSet(
+              sources = setOf<URI>(),
+              sourceRoots = setOf<URI>()
+          ),
+          resources = setOf<URI>(),
+          outputs = setOf<URI>(),
+          sourceDependencies = setOf<URI>(),
+          languageData = rustModule,
+          environmentVariables = mapOf<String, String>()
+      )
+
+  private fun createRustModule(
+      crateId: String,
+      crateRoot: String,
+      location: RustCrateLocation = RustCrateLocation.EXEC_ROOT,
+      crateFeatures: List<String> = emptyList(),
+      dependencies: List<RustDependency>,
+      procMacroArtifacts: List<String> = emptyList()
+  ): RustModule =
+      RustModule(
+          crateId = crateId,
+          location = location,
+          fromWorkspace = true,
+          name = "name", // TODO where is it used?
+          kind = "kind",
+          edition = "edition",
+          crateFeatures = crateFeatures,
+          dependencies = dependencies,
+          crateRoot = crateRoot,
+          version = "version",
+          procMacroArtifacts = procMacroArtifacts,
+          procMacroSrv = "/path/to/procMacroSrv",
+          rustcSysroot = "/path/to/rustcSysroot",
+          rustcSrcSysroot = "/path/to/rustcSrcSysroot",
+          cargoBinPath = "/path/to/cargoBinPath",
+          rustcVersion = "rustcVersion",
+          rustcHost = "/path/to/rustcHost",
```

Comment:
There is no 'path' to the host. 
Rustc host can be e.g. `x86_64-unknown-linux-gnu`